### DISCUSSION TITLE
Several embedding without duplicating source

### DISCUSSION THREAD

**Discussion by jrabary** [lancedb/lancedb#1211] [General]
Hi,

The current method for creating embeddings is relatively straightforward but requires the "source" column (e.g., image) to be in the same table as the embedding vector:

```python
class Pets(LanceModel):
    vector: Vector(clip.ndims) = clip.VectorField()
    image_uri: str = clip.SourceField()
```

In our application, we have several types of embeddings that can be computed at any time. Additionally, the source is located in a separate table since it also has its own metadata.

To avoid source duplication, it would be beneficial if we could use only the "source_id" instead of the "uri" as the source field when computing embeddings, and then perform a join operation to retrieve the actual source data when needed.

Is this feasible with the current state of LanceDB?

**Comment by AyushExel** [comment]
Yeah that is possible. Essentially, you can accept any type of input as source depending on how you design your custom embedding function. See details here - https://lancedb.github.io/lancedb/embeddings/custom_embedding_function/
